src/emcpy/plots/map_tools.py

```diff
@@ -0,0 +1,127 @@
+                                        0, 60, 120, 180))
+        self.yticks = dd.get('yticks', (-90, -60, -30, 0,
+                                        30, 60, 90))
+
+    def _north_america(self, dd=dict()):
+        """
+        Sets extent, longitude xticks, and latitude yticks
+        for a north american domain.
+        """
+        self.extent = (-170, -50, 7.5, 75)
+        self.xticks = dd.get('xticks', (-170, -150, -130, -110,
+                                        -90, -70, -50))
+        self.yticks = dd.get('yticks', (10, 30, 50, 70))
+
+    def _conus(self, dd=dict()):
+        """
+        Sets extent, longitude xticks, and latitude yticks
+        for a contiguous United States domain.
+        """
+        self.extent = (-130, -65, 20, 55)
+        self.xticks = dd.get('xticks', (-130, -120, -110, -100,
+                                        -90, -80, -70, -60))
+        self.yticks = dd.get('yticks', (25, 35, 45, 55))
+
+    def _europe(self, dd=dict()):
+        """
+        Sets extent, longitude xticks, and latitude yticks
+        for a European domain.
+        """
+        self.extent = (-12.5, 40, 30, 70)
+        self.xticks = dd.get('xticks', (-10, 0, 10, 20, 30, 40))
+        self.yticks = dd.get('yticks', (30, 40, 50, 60, 70))
+
+    def _custom(self, dd=dict()):
+        """
+        Sets extent, longitude xticks, and latitude yticks
+        for a Custom domain.
+        """
+        self.extent = dd.extent
+        self.xticks = dd.xticks
+        self.yticks = dd.yticks
+
+
+class MapProjection:
+
+    def __init__(self, projection='plcarr',
+                 cenlon=0,
+                 cenlat=0,
+                 globe=None):
+
+        self.cenlon = cenlon
+        self.cenlat = cenlat
+        self.globe = globe
+
+        map_projections = {
+            "plcarr": self._platecarree,
+            "mill": self._miller,
+            "npstere": self._npstereo,
+            "spstere": self._spstereo}
+
+        try:
+            map_projections[projection]()
+        except KeyError:
+            raise TypeError(f'{domain} is not a valid domain.' +
+                            'Current domains supported are:\n' +
+                            f'{" | ".join(map_projections.keys())}"')
+
+    def _platecarree(self):
+        """Creates projection using PlateCarree from Cartopy."""
+        self.projection = ccrs.PlateCarree(central_longitude=self.cenlon,
+                                           globe=self.globe)
+
+    def _miller(self):
+        """Creates projection using Miller from Cartopy."""
+        self.projection = ccrs.Miller(central_longitude=self.cenlon,
+                                      globe=self.globe)
+
+    def _npstereo(self):
+        """
+        Creates projection using Orthographic from Cartopy and
+        orients it from central latitude 90 degrees.
+        """
+        self.cenlat = 90 if self.cenlat == 0 else self.cenlat
```

Comment:
This is a problem right,
what if someone puts `self.cenlat = 1`.
The projection is not NP Stereographic.